// @progress/kendo-react-layout/dist/npm/breadcrumb/BreadcrumbDelimiter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var PropTypes = require("prop-types");
var kendo_licensing_1 = require("@progress/kendo-licensing");
var kendo_react_common_1 = require("@progress/kendo-react-common");
var package_metadata_1 = require("../package-metadata");
/**
 * Represents the BreadcrumbDelimiter component.
 */
exports.BreadcrumbDelimiter = React.forwardRef(function (props, ref) {
    kendo_licensing_1.validatePackage(package_metadata_1.packageMetadata);
    var target = React.useRef(null);
    var breadcrumbDelimiterRef = React.useRef(null);
    React.useImperativeHandle(target, function () { return ({
        element: breadcrumbDelimiterRef.current,
        props: props
    }); });
    React.useImperativeHandle(ref, function () { return target.current; });
    return (React.createElement("span", { ref: breadcrumbDelimiterRef, id: props.id, tabIndex: props.tabIndex, style: props.style, className: kendo_react_common_1.classNames('k-icon k-i-arrow-chevron-right k-breadcrumb-delimiter-icon', props.className) }));
});
var propTypes = {
    id: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object,
    tabIndex: PropTypes.number
};
exports.BreadcrumbDelimiter.displayName = 'KendoReactBreadcrumbDelimiter';
// TODO: delete casting when @types/react is updated!
exports.BreadcrumbDelimiter.propTypes = propTypes;
