// @progress/kendo-react-layout/dist/npm/breadcrumb/BreadcrumbListItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var PropTypes = require("prop-types");
var kendo_licensing_1 = require("@progress/kendo-licensing");
var kendo_react_common_1 = require("@progress/kendo-react-common");
var package_metadata_1 = require("../package-metadata");
/**
 * Represents the BreadcrumbListItem component.
 */
exports.BreadcrumbListItem = React.forwardRef(function (props, ref) {
    kendo_licensing_1.validatePackage(package_metadata_1.packageMetadata);
    var target = React.useRef(null);
    var breadcrumbLIRef = React.useRef(null);
    var focus = React.useCallback(function () {
        if (breadcrumbLIRef.current) {
            breadcrumbLIRef.current.focus();
        }
    }, [breadcrumbLIRef]);
    React.useImperativeHandle(target, function () { return ({
        element: breadcrumbLIRef.current,
        focus: focus,
        props: props
    }); });
    React.useImperativeHandle(ref, function () { return target.current; });
    return (React.createElement("li", { ref: breadcrumbLIRef, id: props.id, style: props.style, className: kendo_react_common_1.classNames('k-flex-none k-breadcrumb-item', props.className) }, props.children));
});
var propTypes = {
    id: PropTypes.string,
    className: PropTypes.string,
    children: PropTypes.any,
    style: PropTypes.object
};
exports.BreadcrumbListItem.displayName = 'KendoReactBreadcrumbListItem';
// TODO: delete casting when @types/react is updated!
exports.BreadcrumbListItem.propTypes = propTypes;
