// @progress/kendo-react-layout/dist/npm/breadcrumb/BreadcrumbOrderedList.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var PropTypes = require("prop-types");
var kendo_licensing_1 = require("@progress/kendo-licensing");
var kendo_react_common_1 = require("@progress/kendo-react-common");
var package_metadata_1 = require("../package-metadata");
/**
 * Represents the BreadcrumbOrderedList component.
 */
exports.BreadcrumbOrderedList = React.forwardRef(function (props, ref) {
    kendo_licensing_1.validatePackage(package_metadata_1.packageMetadata);
    var target = React.useRef(null);
    var orderedListRef = React.useRef(null);
    React.useImperativeHandle(target, function () { return ({
        element: orderedListRef.current,
        props: props
    }); });
    React.useImperativeHandle(ref, function () { return target.current; });
    return (React.createElement("ol", { id: props.id, ref: orderedListRef, style: props.style, dir: kendo_react_common_1.useDir(orderedListRef, props.dir), tabIndex: kendo_react_common_1.getTabIndex(props.tabIndex, props.disabled), className: kendo_react_common_1.classNames('k-breadcrumb-container k-flex-wrap', {
            'k-state-disabled': props.disabled
        }, props.className) }, props.children));
});
var propTypes = {
    id: PropTypes.string,
    className: PropTypes.string,
    children: PropTypes.element,
    tabIndex: PropTypes.number,
    style: PropTypes.object,
    dir: PropTypes.string,
    disabled: PropTypes.bool
};
exports.BreadcrumbOrderedList.displayName = 'KendoReactBreadcrumbOrderedList';
// TODO: delete casting when @types/react is updated!
exports.BreadcrumbOrderedList.propTypes = propTypes;
