// @progress/kendo-react-layout/dist/es/breadcrumb/BreadcrumbDelimiter.js
import * as React from 'react';
import * as PropTypes from 'prop-types';
import { validatePackage } from '@progress/kendo-licensing';
import { classNames } from '@progress/kendo-react-common';
import { packageMetadata } from '../package-metadata';
/**
 * Represents the BreadcrumbDelimiter component.
 */
export var BreadcrumbDelimiter = React.forwardRef(function (props, ref) {
    validatePackage(packageMetadata);
    var target = React.useRef(null);
    var breadcrumbDelimiterRef = React.useRef(null);
    React.useImperativeHandle(target, function () { return ({
        element: breadcrumbDelimiterRef.current,
        props: props
    }); });
    React.useImperativeHandle(ref, function () { return target.current; });
    return (React.createElement("span", { ref: breadcrumbDelimiterRef, id: props.id, tabIndex: props.tabIndex, style: props.style, className: classNames('k-icon k-i-arrow-chevron-right k-breadcrumb-delimiter-icon', props.className) }));
});
var propTypes = {
    id: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object,
    tabIndex: PropTypes.number
};
BreadcrumbDelimiter.displayName = 'KendoReactBreadcrumbDelimiter';
// TODO: delete casting when @types/react is updated!
BreadcrumbDelimiter.propTypes = propTypes;
